diesel/src/sqlite/types/date_and_time/chrono.rs

```diff
@@ -98,19 +100,90 @@ impl ToSql<Timestamp, Sqlite> for NaiveDateTime {
     }
 }
 
+impl FromSql<TimestampWithTz, Sqlite> for NaiveDateTime {
+    fn from_sql(value: backend::RawValue<'_, Sqlite>) -> deserialize::Result<Self> {
+        value.parse_string(|text| {
+            let sqlite_datetime_formats = &[
+                "%F %RZ",
+                "%F %R%:z",
+                "%F %TZ",
+                "%F %T%:z",
+                "%F %T%.fZ",
+                "%F %T%.f%:z",
+                "%FT%RZ",
+                "%FT%R%:z",
+                "%FT%TZ",
+                "%FT%T%:z",
+                "%FT%T%.fZ",
+                "%FT%T%.f%:z",
+            ];
+
+            for format in sqlite_datetime_formats {
+                if let Ok(dt) = Self::parse_from_str(text, format) {
+                    return Ok(dt);
+                }
+            }
+
+            if let Ok(julian_days) = text.parse::<f64>() {
+                let epoch_in_julian_days = 2_440_587.5;
+                let seconds_in_day = 86400.0;
+                let timestamp = (julian_days - epoch_in_julian_days) * seconds_in_day;
+                let seconds = timestamp as i64;
+                let nanos = (timestamp.fract() * 1E9) as u32;
+                if let Some(timestamp) = Self::from_timestamp_opt(seconds, nanos) {
+                    return Ok(timestamp);
+                }
+            }
+
+            Err(format!("Invalid datetime {}", text).into())
+        })
+    }
+}
+
+impl ToSql<TimestampWithTz, Sqlite> for NaiveDateTime {
+    fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, Sqlite>) -> serialize::Result {
+        out.set_value(self.format("%F %T%.f%:z").to_string());
+        Ok(IsNull::No)
+    }
+}
+
+impl FromSql<TimestampWithTz, Sqlite> for DateTime<Utc> {
+    fn from_sql(value: backend::RawValue<'_, Sqlite>) -> deserialize::Result<Self> {
+        let naive_date_time = <NaiveDateTime as FromSql<TimestampWithTz, Sqlite>>::from_sql(value)?;
+        Ok(DateTime::from_utc(naive_date_time, Utc))
+    }
+}
+
+impl FromSql<TimestampWithTz, Sqlite> for DateTime<Local> {
+    fn from_sql(value: backend::RawValue<'_, Sqlite>) -> deserialize::Result<Self> {
+        let naive_date_time = <NaiveDateTime as FromSql<TimestampWithTz, Sqlite>>::from_sql(value)?;
+        Ok(Local::from_utc_datetime(&Local, &naive_date_time))
+    }
+}
+
+impl<TZ: TimeZone> ToSql<TimestampWithTz, Sqlite> for DateTime<TZ> {
+    fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, Sqlite>) -> serialize::Result {
+        out.set_value(self.naive_utc().format("%F %T%.f%:z").to_string());
+        Ok(IsNull::No)
+    }
+}
```

Comment:
Based on the documentation for `ToSql` I was led to believe that I should be trying to reuse an existing implementation (in this case I was attempting to reuse `ToSql<TimestampWithTz, Sqlite> for NaiveDateTime`) but I was getting the following errors I think due to differences in how exactly these parameters bind to the query engine itself (I don't really know exactly what I'm talking about but found the below code snippet interesting and will be looking it up)

https://github.com/diesel-rs/diesel/blob/7184f128a3c1435d12b32f413b14b6632e2f753c/diesel/src/serialize.rs#L165-L170

```
impl<TZ: TimeZone> ToSql<TimestampWithTz, Sqlite> for DateTime<TZ> {
    fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, Sqlite>) -> serialize::Result {
        <ToSql<TimestampWithTz, Sqlite>>::to_sql(&self.naive_utc(), &mut out.reborrow())
    }
}

...

error[E0599]: the method `reborrow` exists for mutable reference `&mut serialize::Output<'b, '_, Sqlite>`, but its trait bounds were not satisfied
   --> diesel/src/sqlite/types/date_and_time/chrono.rs:166:78
    |
166 |         <ToSql<TimestampWithTz, Sqlite>>::to_sql(&self.naive_utc(), &mut out.reborrow())
    |                                                                              ^^^^^^^^ method cannot be called on `&mut serialize::Output<'b, '_, Sqlite>` due to unsatisfied trait bounds
    |
   ::: diesel/src/sqlite/backend.rs:10:1
    |
10  | pub struct Sqlite;
    | ------------------ doesn't satisfy `<_ as backend::private::HasBindCollector<'a>>::BindCollector = RawBytesBindCollector<Sqlite>`
    |
    = note: the following trait bounds were not satisfied:
            `<Sqlite as backend::private::HasBindCollector<'a>>::BindCollector = RawBytesBindCollector<Sqlite>`
```